tests/tensors/test_bfv_vector.py

```diff
@@ -320,18 +418,28 @@ def test_mul_plain(context, vec1, vec2):
         ([1, 2], [-73, -10]),
     ],
 )
-def test_mul_plain_inplace(context, vec1, vec2):
+@pytest.mark.parametrize(
+    "duplicate", [noop, deep_copy, simple_copy, internal_copy, recreate, pickled,]
+)
+def test_mul_plain_inplace(context, vec1, vec2, duplicate):
     first_vec = ts.bfv_vector(context, vec1)
+    first_vec = duplicate(first_vec)
+
     second_vec = vec2
     first_vec *= second_vec
+    first_vec = duplicate(first_vec)
+
     expected = [v1 * v2 for v1, v2 in zip(vec1, vec2)]
 
     # Decryption
     decrypted_result = first_vec.decrypt()
     assert decrypted_result == expected, "Multiplication of vectors is incorrect."
 
 
-def test_size(context):
+@pytest.mark.parametrize(
+    "duplicate", [noop, deep_copy, simple_copy, internal_copy, recreate, pickled,]
+)
+def test_size(context, duplicate):
     for size in range(10):
         vec = ts.bfv_vector(context, [1] * size)
         assert vec.size() == size, "Size of encrypted vector is incorrect."
```

Comment:
While I like the idea of integrating the serialization tests into the others, it's a lot to process for tests, I think adding separate test for the general use cases can cover the same functionalities while reducing tests time